### ISSUE TITLE
"Cannot convert tz-naive timestamps"

### ISSUE THREAD

**Issue by vfilimonov** [pydata/pandas#9865]
I'm not sure if it is an issue of DOCs or code.

[Docs](http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.tz_convert.html) say that `tz_convert` "Convert the axis to target time zone. *If it is time zone naive, it will be localized to the passed time zone*."

But the code in contrast raises an [exception](https://github.com/pydata/pandas/blob/bb9d35c27abdd68f349950c404292426f730759d/pandas/tseries/index.py#L1597):
```
tmp = pd.Series([0], index=[pd.Timestamp('2008-12-16 13:10:00')])
tmp.tz_convert('UTC')
```
Results in `TypeError: Cannot convert tz-naive timestamps, use tz_localize to localize`

**Comment by jorisvandenbossche** [comment]
I think it is a docs thing. Also the docs on `DatetimeIndex.tz_convert` are correct in that way: http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DatetimeIndex.tz_convert.html#pandas.DatetimeIndex.tz_convert

**Comment by jorisvandenbossche** [comment]
@jreback correct?

**Comment by jreback** [comment]
yeh, this is just an old/not updated doc-string. This will only *convert* if it is already non-naive, while ``tz_localize`` only will work if its naive. This is the original API, and was separate on purpose to prevent non-explict conversions.

want to do a PR to update the doc-strings (maybe add a ``Raises`` section to both of these functions in ``tseries/index.py`` as well as the functions in ``core/generic.py``

**Comment by vfilimonov** [comment]
Sure, will do docs.

@jreback , what do you mean by `Raises` section?

**Comment by jorisvandenbossche** [comment]
A section in the docstring, see https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections

**Comment by vfilimonov** [comment]
OK, will do

**Comment by jorisvandenbossche** [comment]
Closed by #9906
